custom_connectors/oauth2/miro.rb

```diff
@@ -0,0 +1,232 @@
+      end,
+      output_fields: lambda do |object_definitions|
+        object_definitions['board']
+      end
+    },
+
+    create_card_widget: {
+      description: 'Creates a Card Widget on board in Miro',
+      input_fields: lambda do
+        [
+          {
+            name: 'board',
+            control_type: 'select',
+            pick_list: 'boards',
+            optional: false
+          },
+          {
+            name: 'frame',
+            control_type: 'select',
+            pick_list: 'frames',
+            optional: false,
+            pick_list_params: { board: 'board' },
+            hint: 'Switch frame to grid mode to avoid cards overlap'
+          },
+          { name: 'title' },
+          { name: 'link', label: 'Title link', control_type: 'url' },
+          { name: 'description' },
+          { name: 'border_color', hint: 'In hex format (default is #2399f3)' },
+          { name: 'due_date', type: 'date', control_type: 'date' }
+        ]
+      end,
+
+      execute: lambda do |_connection, input|
+        payload = { type: 'card', parentFrameId: input['frame'] }
+        title = input['title']
+        if title.present?
+          payload[:title] = title
+          link = input['link']
+          if link.present?
+            payload[:title] =
+              "<p><a href=\"#{link}\" target=\"_blank\">#{title}</a></p>"
+          end
+        end
+        if input['description'].present?
+          payload[:description] = input['description']
+        end
+        if input['border_color'].present?
+          payload[:style] = { backgroundColor: input['border_color'] }
+        end
+        if input['due_date'].present?
+          payload[:dueDate] = {
+            dueDate: [input['due_date'].strftime('%Q').to_i]
+          }
+        end
+        post("/v1/boards/#{input['board']}/widgets")
+          .payload(payload)
+      end,
+
+      output_fields: lambda do |object_definitions|
+        object_definitions['card']
+      end
+    }
+  },
+
+  triggers: {},
+
+  pick_lists: {
+
+    board_access_by_link: lambda do
+      [
+        ['Private — only you have access to the board', 'private'],
+        ['View — can view, no sign-in required', 'view'],
+        ['Comment — can comment, no sign-in required', 'comment']
+      ]
+    end,
+
+    board_access_within_account: lambda do
+      [
+        ['Private — nobody in the team can access this board', 'private'],
+        ['View — any team member can find and view the board', 'view'],
+        ['Comment — any team member can find and comment the board', 'comment'],
+        ['Edit — any team member can find and edit the board', 'edit']
+      ]
+    end,
+
+    boards: lambda do
+      account_id = get('/v1/oauth-token')['account']['id']
+      query = 'fields=id,name&limit=500'
+      boards_resp = get("v1/accounts/#{account_id}/boards?#{query}")
+      next_link = boards_resp['nextLink']
+      boards = boards_resp['data']
+
+      while next_link.present?
```

Comment:
How many boards do 80% of your users have? thats normally a good benchmark for how many boards to return in a picklist

Reply:
We normally don't encourage loops when building out picklists